wagtail/wagtailcore/blocks/field_block.py

```diff
@@ -326,36 +331,58 @@ def __init__(self, choices=None, required=True, help_text=None, **kwargs):
-                if has_blank_choice:
-                    break
-            else:
-                # this is an individual choice; v1 is the value
-                if v1 in ('', None):
-                    has_blank_choice = True
-                    break
-
-        if not has_blank_choice:
-            choices = BLANK_CHOICE_DASH + choices
-
+        choices = self.get_callable_choices(choices)
         self.field = forms.ChoiceField(choices=choices, required=required, help_text=help_text)
         super(ChoiceBlock, self).__init__(**kwargs)
 
+    def get_callable_choices(self, choices):
+        """
+        Because we need to support callable choices and insert blank choice into choices list
+        we will always pass callable into `forms.ChoiceField`.
+        """
+        def choices_callable():
+            # Variable choices can be an instance of CallableChoiceIterator,
+            # and we don't want to call it twice if produced list doesn't contain blank choice,
+            # so cast it as a list here
+            local_choices = list(choices)
+
+            # If choices does not already contain a blank option, insert one
+            # (to match Django's own behaviour for modelfields:
+            # https://github.com/django/django/blob/1.7.5/django/db/models/fields/__init__.py#L732-744)
+            has_blank_choice = False
+            for v1, v2 in local_choices:
+                if isinstance(v2, (list, tuple)):
+                    # this is a named group, and v2 is the value list
+                    has_blank_choice = any([value in ('', None) for value, label in v2])
+                    if has_blank_choice:
+                        break
+                else:
+                    # this is an individual choice; v1 is the value
+                    if v1 in ('', None):
+                        has_blank_choice = True
+                        break
+
+            if not has_blank_choice:
+                return BLANK_CHOICE_DASH + list(local_choices)
+
+            return local_choices
+        return choices_callable
+
     def deconstruct(self):
         """
         Always deconstruct ChoiceBlock instances as if they were plain ChoiceBlocks with their
         choice list passed in the constructor, even if they are actually subclasses. This allows
         users to define subclasses of ChoiceBlock in their models.py, with specific choice lists
         passed in, without references to those classes ending up frozen into migrations.
         """
-        return ('wagtail.wagtailcore.blocks.ChoiceBlock', [], self._constructor_kwargs)
+
+        # Actually, we can just return CallableChoiceIterator,
+        # but it will be casted as a list in __init__ method
+        _constructor_kwargs = self._constructor_kwargs.copy()
+        if isinstance(_constructor_kwargs['choices'], CallableChoiceIterator):
+            _constructor_kwargs['choices'] = _constructor_kwargs['choices'].choices_func
```

Comment:
Probably we don't need this, if we are not going to support callable choices using subclassing... Needs extra attention while code review, anyway.